client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -713,15 +713,25 @@ private TopicPublishInfo tryToFindTopicPublishInfo(final String topic) {
         TopicPublishInfo topicPublishInfo = this.topicPublishInfoTable.get(topic);
         if (null == topicPublishInfo || !topicPublishInfo.ok()) {
             this.topicPublishInfoTable.putIfAbsent(topic, new TopicPublishInfo());
-            this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic);
             topicPublishInfo = this.topicPublishInfoTable.get(topic);
+            if (!topicPublishInfo.checkNotFoundFlag()) {
+                this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic);
+                topicPublishInfo = this.topicPublishInfoTable.get(topic);
+            }
         }
 
         if (topicPublishInfo.isHaveTopicRouterInfo() || topicPublishInfo.ok()) {
             return topicPublishInfo;
         } else {
-            this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic, true, this.defaultMQProducer);
-            topicPublishInfo = this.topicPublishInfoTable.get(topic);
+            boolean found;
+            if (!topicPublishInfo.checkNotFoundFlag()) {
+                found = this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic, true, this.defaultMQProducer);
+                topicPublishInfo = this.topicPublishInfoTable.get(topic);
+                if (!found) {
+                    topicPublishInfo.setNotFoundFlag();
```

Comment:
Is it appropriate to set the origin topic not found flag here？

Reply:
Yes, the NotFound state should be cached if and only if the original topic and TBW102 are not found.